Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -211,9 +211,31 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
     Cookie cookie;
     cookie.name = WTFMove(options.name);
     cookie.value = WTFMove(options.value);
-    cookie.domain = options.domain.isNull() ? document.domain() : WTFMove(options.domain);
     cookie.created = WallTime::now().secondsSinceEpoch().milliseconds();
 
+    cookie.domain = options.domain.isNull() ? document.domain() : WTFMove(options.domain);
+    auto& url = document.url();
+    if (!cookie.domain.isNull()) {
+        if (cookie.domain.startsWith('.')) {
+            promise->reject(Exception { TypeError, "The domain must not begin with a '.'"_s });
+            return;
+        }
+
+        auto host = url.host();
+        if (host != cookie.domain) {
+            if (host.length() <= cookie.domain.length() || host.characterAt(host.length() - cookie.domain.length() - 1) != '.' || !host.endsWith(cookie.domain)) {
+                promise->reject(Exception { TypeError, "The domain must be a part of the current host"_s });
+                return;
+            }
+        }
+
+        static constexpr auto maximumAttributeValueSize = 1024;
```

Comment:
Could you add a spec link in a comment for this?  Otherwise it looks like just a magic number.